physics/mynnedmf_wrapper.F90

```diff
@@ -588,21 +540,38 @@ SUBROUTINE mynnedmf_wrapper_run(        &
          allocate(old_ozone(im,levs))
          old_ozone = ozone
        endif
-       if (lprnt)write(0,*)"prepping MYNN-EDMF variables..."
 
        do k=1,levs
           do i=1,im
-          !   dz(i,k)=(phii(i,k+1) - phii(i,k))*g_inv
              th(i,k)=t3d(i,k)/exner(i,k)
-          ! keep as specific humidity
-          !   qv(i,k)=qvsh(i,k)/(1.0 - qvsh(i,k))
-          !   qc(i,k)=qc(i,k)/(1.0 - qvsh(i,k))
-          !   qi(i,k)=qi(i,k)/(1.0 - qvsh(i,k))
-             rho(i,k)=prsl(i,k)/(r_d*t3d(i,k))
+             rho(i,k)=prsl(i,k)/(r_d*t3d(i,k)*(1.+p608*max(sqv(i,k),1e-8)))
              w(i,k) = -omega(i,k)/(rho(i,k)*grav)
+          enddo
+       enddo
+
+  ! Check incoming moist species to ensure non-negative values
+  ! First, create height difference (dz)
+      do k=1,levs
+         do i=1,im
+            dz(i,k)=(phii(i,k+1) - phii(i,k))*g_inv
```

Comment:
It's amazing to me that `dz` and `delp` don't already exist as CCPP variables to pass in. It looks like RRTMGP calculates/saves them, but other schemes, if they do calculate them (and I'm sure that some do), they're not sharing them.

Reply:
Typically, each physics scheme calculate dz between full levels, and between half levels, dp between full levels when the scheme needs them. Keep in mind that geopotential height is updated in CCPP.

Reply:
Yep, all good points, and good reason to recalculate as necessary in case the geopotential update ever changes order within the suite.

Reply:
I think this is just a comment, but yes, perhaps there is a small case for some redundancy.